source/extensions/network/dns_resolver/getaddrinfo/getaddrinfo.cc

```diff
@@ -154,10 +163,27 @@ void GetAddrInfoDnsResolver::resolveThreadRoutine() {
       if (rc.return_value_ == 0) {
         response = processResponse(*next_query, addrinfo_wrapper.get());
       } else if (reresolve && rc.return_value_ == EAI_AGAIN) {
-        ENVOY_LOG(debug, "retrying query [{}]", next_query->dns_name_);
         absl::MutexLock guard(&mutex_);
-        pending_queries_.push_back(next_query);
-        continue;
+        if (num_retries.has_value()) {
+          (*num_retries)--;
+        }
+        if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.getaddrinfo_num_retries") &&
```

Comment:
do we need to check the runtime guard here again?  isn't the presence of nullopt the indication that the feature isn't set?

Reply:
Done.

Reply:
Yeah the presence of `nullopt` indicates that the feature isn't set. I updated the code to remove unnecessary runtime guard checks.